src/pudl_rmi/coordinate.py

```diff
@@ -166,19 +225,35 @@ def ferc1_to_eia(self, clobber=False, clobber_plant_parts_eia=False):
             clobber_plant_parts_eia (boolean): Generate and cache a new interim
                 output of the EIA plant part list and generate a new version of
                 the depreciaiton to FERC1 output. Default is False.
+            clobber_plant_parts_eia_distinct(boolean): Generate and cache a new
+                output of the distinct EIA plant part list
         """
         file_path = pudl_rmi.FERC1_EIA_PKL
         # if any of the clobbers are on, we want to regenerate the main output
-        clobber_any = any([clobber, clobber_plant_parts_eia])
+        clobber_any = any(
+            [clobber, clobber_plant_parts_eia, clobber_plant_parts_eia_distinct]
+        )
         check_is_file_or_not_exists(file_path)
         if not file_path.exists() or clobber_any:
             logger.info(
                 f"FERC to EIA granular connection not found at {file_path}... "
                 "Generating a new output."
             )
+            # get or generate connected training data
+            train_file_path = pudl_rmi.CONNECTED_TRAIN_PKL
```

Comment:
Moved this connection of training data to EIA true granularity records here - before `connect_ferc1_to_eia.execute` is called.